build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -520,6 +520,49 @@ message Platform {
   // The server MAY use the `value` of one or more properties to determine how
   // it sets up the execution environment, such as by making specific system
   // files available to the worker.
+  //
+  // The following standard `name`s are defined:
+  //
+  // * `OS`: This describes the operating system. Only a single value is allowed
```

Comment:
Is your intention that all servers *must* support OS? If not, how about adding a caveat:
  "Only a single value..." -> "If supported, only a single value..." ?

Reply:
I wouldn't mandate it, however, I think it would make sense for most servers to support it. The only exception being servers that don't accept any OS-specific input.

From a client request perspective, "If supported, only a single value..." seems a bit odd, in my opinion. Is this just me? How about "Multiple values are not allowed." as alternative?

Reply:
"Multiple values are not allowed" sounds better, yeah, thanks!

And yeah, it being _recommended_ especially for multi-OS services sounds appropriate to me, but this way there's a nice carveout for services backing homogeneous fleets and other such use-cases.

Reply:
"it would make sense for most servers to support it" is great in general. Unfortunately, I'm not sure if we'll get there just by adding a comment to the API.

Actually, do we know the state of the world wrt this?

I understand BuildBarn has no immediate plans to support routing based on Platform and it would be a lot of work to bring it to conformance; RBE has no near-term plans for this type of matching.
Do we know the state of support/near-term plans for the other servers?

Reply:
@bwkimmel to comment on RBE's interest and timeline in supporting OS specifically.

Reply:
> I understand BuildBarn has no immediate plans to support routing based on Platform and it would be a lot of work to bring it to conformance

A possible option for minimal support would be to at least reject incompatible requests, even if only a single worker platform is supported per instance.

Reply:
Perhaps rename to `OSFamily`? `OS` seems a bit ambiguous regarding how specific this ought to be. I.e., should it only be the high-level `OS` "family" as you've specified here? Or should it represent the full specification (`uname -a`), or something in between.

Internally, RBE has been using `OSFamily` to represent this value (i.e., just `Linux`, `Windows`, etc.) -- though as Eric indicated, this is not yet supported externally. We could later standardize other variables to indicate more specific values, e.g., `OSDistribution` might take on values like `Ubuntu`, `Debian`, etc., `OSDistributionVersion` for the version number of that distribution (e.g., `18.04`).

Reply:
`OSFamily` sounds fine to me.

Reply:
I've changed this to `OSFamily`. Does this look ok now?